Docs: mention gRPC compression in "Production tips" (#3201)
grafana/mimir: docs/sources/operators-guide/run-production-environment/production-tips/index.md

We recommend securing the Grafana Mimir cluster.
For more information about securing a Mimir cluster, refer to [Secure Grafana Mimir]({{< relref "../../secure/_index.md" >}}).

## Network

Most of the communication between Mimir components occurs over gRPC. The gRPC
connection does not use any compression by default. This can be most noticeable in the connection between a
distributor and an ingester.

If network throughput is a concern or a high cost, then you can enable compression on the gRPC connection between 
components. This will reduce the required throughput at the cost of CPU usage. You can choose between gzip and 

> snappy. Gzip provides better compression than snappy at the cost of more CPU usage.

Reviewer: Can we link to some sort of representative benchmarks here?

We recommend securing the Grafana Mimir cluster.
For more information about securing a Mimir cluster, refer to [Secure Grafana Mimir]({{< relref "../../secure/_index.md" >}}).

## Network

Most of the communication between Mimir components occurs over gRPC. The gRPC
connection does not use any compression by default. This can be most noticeable in the connection between a

> distributor and an ingester.

Reviewer 2: Reason is explained here: https://github.com/grafana/mimir/blob/main/docs/sources/operators-guide/configure/configuring-zone-aware-replication.md?plain=1#L57-L59, if you want to include it.

Author: I'd fall back to Ursula - is this necessary detail? It does not give any actionable information to the user. But will give an explanation if they are curious. I am not sure how curious our users are.

> connection does not use any compression by default. This can be most noticeable in the connection between a
> distributor and an ingester.

Reviewer 3: ```suggestion
connection does not use any compression by default. This can be most noticeable is the connection between
distributors and an ingesters.
```

> connection does not use any compression by default. This can be most noticeable is the connection between
> distributors and an ingesters.

Reviewer 4: ```suggestion
connection does not use any compression by default.
```

If network throughput is a concern or a high cost, then you can enable compression on the gRPC connection between

> components. This will reduce the required throughput at the cost of CPU usage. You can choose between gzip and

Reviewer 3: ```suggestion
components. This will reduce the network throughput at the cost of increased CPU usage. You can choose between gzip and
```

snappy. Gzip provides better compression than snappy at the cost of more CPU usage.

You can use the [Squash Compression Benchmark](http://quixdb.github.io/squash-benchmark/#results-table) to choose between snappy and gzip.
For protobuf data snappy achieves a compression ratio of 5 with compression speeds of
around 400MiB/s. For the same data gzip achieves a ratio between 6 and 8 with speeds between 50MiB/s and 135 MiB/s.

> To configure gRPC compression use the following CLI flags or their YAML equivalents. The accepted values are

Reviewer 2: Also empty string is accepted value, but it disables compression.

> `snappy` and `gzip`. Setting the flag to an empty string (`''`) explicitly disables compression.

Reviewer 4: ```suggestion
`snappy` and `gzip`. If you set the flag to an empty string (`''`), it explicitly disables compression.
```
